src/components/Dashboard/StudentData/index.js

```diff
@@ -69,7 +69,7 @@ function StudentData() {
           "partner",
         ])
       ) {
-        if (res.data.students.length < 1) {
+        if (res.data.students && res?.data?.students.length < 1) {
```

Comment:
Yahan, `res.data` object hona chahiye (https://axios-http.com/docs/res_schema) toh `res` yah `data` ke baad `?` zarooree nahi hai. Otherwise, using `res.data.students` would cause the program to crash in these instances. Also, we need to set the message to "There are no results to display" when `res.data.students` is `undefined` or we'll get a new error when it tries to read the `map` property in the `else` statement.
```suggestion
        if (!Array.isArray(res.data.students) || res.data.students.length === 0) {
```
Lekin kyon `res.data` ke paas `student` property kabhi kabhi nahi hoga?